crates/containerd-shim-wasm/src/sandbox/shim.rs

```diff
@@ -39,6 +39,8 @@ use nix::unistd::mkdir;
 use oci_spec::runtime;
 use ttrpc::context::Context;
 
+static CONTAINER_ROOT_DIR: &str = "/var/run/runwasi";
```

Comment:
I can't help but think this should be specified by the implementation, ie `/run/io.containerd.wasmedge.v1` OR from the spec?

Reply:
It's also not clear to me what this is used for

Reply:
This directory is the same role as runc; it stores files for managing the state of the container and signaling start.
If runwasi is to be considered only as an OCI runtime, I don't think the name containerd should be included in the directory. I think that including it may compromise its versatility.
https://github.com/opencontainers/runc/blob/efad7a3b80f4585065f93aad3e91989da5012d63/man/runc.8.md?plain=1#L108-L111

Reply:
In our case containerd is passing down a state dir to the shim that we should be using rather than coming up with our own.

I can't exactly recall how we setting that dir right now but since this crate is only a library I think we should leave any sort of state dir to a higher level.

Reply:
@cpuguy83 Thanks for your comment 🙇 That is sort of an ideal, but when it comes to adjusting to reality it takes time. Many OCI Runtimes already require a state command. This command is inevitably stateful. Also, it is an OCI specification that `create` command and `start` command are two different commands: `create` creates the container process and the `start` command sent a message to that process to start the container. In other words, the `start` command must leave an interface somewhere to interact with the container created by create. That interface is left in this directory.
https://containers.github.io/youki/user/liboci_cli.html

Reply:
@utam0k Yes I understand that.
For us, containerd passes down the state.
As an example when you create a container with containerd, it's not using the default runc state dir because containerd (or more precisely whoever started the container) is managing that path.

If it's docker `/run/docker/runtime-runc/moby` (where `moby` is the containerd namespace) because docker customizes the path when it invokes containerd. 

```console
# jq . /run/containerd/io.containerd.runtime.v2.task/moby/01e48c19c2af1da405f209b79fcc10542372ab82fb081ad80f713826797e6ded/options.json
{
  "binary_name": "runc",
  "root": "/var/run/docker/runtime-runc",
  "systemd_cgroup": true
}
```

If it's ctr it's `/run/containerd/runc/default/test` (where `default` is the containerd namespace and `test` is the container id).
ctr does not customize this path so it's just the default.

```console
# jq . /run/containerd/io.containerd.runtime.v2.task/default/test/options.json 
{}
```

So we should expect the actual shim implementation to pass down this state dir from higher up the stack.
The above mentioned directories is the OCI bundle path.
So we should be able to check for a state dir in this file and 

We can choose a default path, but it also must be namespaced by the containerd namespace or else we can end up with conflicts.

Reply:
The other part of this is we have a long running daemon managing state and should not need to persist it across calls.

When the daemon dies so do the containers it is managing.

Reply:
@cpuguy83 Make sense a lot 💯 I have updated my PR!